tests/rc_string_ut/rc_string_ut.c

```diff
@@ -54,6 +66,17 @@ TEST_SUITE_INITIALIZE(suite_initialize)
     ASSERT_ARE_EQUAL(int, 0, umock_c_init(on_umock_c_error));
     ASSERT_ARE_EQUAL(int, 0, umocktypes_charptr_register_types());
 
+    REGISTER_GLOBAL_MOCK_HOOK(mocked_vsnprintf, my_vsnprintf);
+    REGISTER_GLOBAL_MOCK_FAIL_RETURN(mocked_vsnprintf, -1);
+
+#ifdef WIN32
+    // The reason this is with a WIN32 block is because 
+    // in linux the va_list item is not a char*, but a struct
+    // of size 24 bytes.  It is made to be opaque so we are unable
+    // to set up the mock alias for this type.
```

Comment:
Interesting, so what if for Linux we're making this alias to a struct of 24 bytes (opaque as it is I assume we can see what it is actually behind in the code)?

Reply:
We can't see what the 24 bytes are and it would possibly change with different version of linux.  Technically we can try to just make the size measure up, but the underlying structure would be unknow.  Would this work with the register?

Reply:
Yeah, but technically you don't need an alias :-D. Just need to define the 3-4 functions that umock needs in order to know and handle a type.